core/src/test/scala/unit/kafka/server/AdvertiseBrokerTest.scala

```diff
@@ -77,17 +80,17 @@ class AdvertiseBrokerTest extends ZooKeeperTestHarness {
     servers += TestUtils.createServer(KafkaConfig.fromProps(props))
 
     val brokerInfo = zkClient.getBroker(brokerId).get
-    assertEquals(1, brokerInfo.endPoints.size)
+    assertEquals(2, brokerInfo.endPoints.size)
     val endpoint = brokerInfo.endPoints.head
     assertEquals("external-listener", endpoint.host)
     assertEquals(9999, endpoint.port)
     assertEquals(SecurityProtocol.PLAINTEXT, endpoint.securityProtocol)
-    assertEquals("EXTERNAL", endpoint.listenerName.value)
+    assertEquals(new ListenerName("EXTERNAL"), endpoint.listenerName)
```

Comment:
This change and the one below is not needed, right?

Reply:
It is, it's comparing a String to a ListenerName otherwise:
```
java.lang.AssertionError: 
Expected :INTERNAL
Actual   :ListenerName(INTERNAL)
```

Reply:
You removed the `value` call from the right hand side. If you revert that change too, then it looks fine to me.